animareflection/ui
# Fix `*.d.ts` generation race condition

## Description

##### Task link: https://trello.com/c/jJ3HW0My/156-dev-performance-optimization

- Fixed race condition caused by `tsup` `onSuccess` hook triggering before `*.d.ts` generation finishes
  - Improved previous solution of flaky sleep syscalls
- Simplified list of external packages for `tsup` build
- Added [dependency cycle](https://en.wikipedia.org/wiki/Circular_dependency) lint rule

## Test Steps

- Verify local build consistently works: CJS & ESM → DTS → local package publish
